security/RFP.md

```diff
@@ -0,0 +1,254 @@
+
+The implementation can be found in this [Github repository](https://github.com/input-output-hk/hydra-poc)
+
+Version to be audited: [0.9.0](https://github.com/input-output-hk/hydra-poc/releases/tag/0.9.0)
+
+TODO describe the inputs and outputs of a hydra node
+
+TODO: clarify which artifacts we need to introduce
+
+TODO:We should share our test architecture/topology and also the test results here: https://hydra.family/head-protocol/benchmarks/tests/hydra-cluster/hspec-results
+
+#### Artifact 3.3: Hydra plutus scripts
+
+TODO
+
+Grab stuff from https://hydra.family/head-protocol/haddock/hydra-plutus/index.html
+
+#### Artifact 3.4: Hydra node chain layer code
+
+TODO
+
+Grab stuff from https://hydra.family/head-protocol/haddock/hydra-node/index.html sub-sections of _Hydra.Chain.Direct_
+
+#### Artifact 3.5: layer 2 code
+
+TODO
+
+## Hypotheses
+
+The Coordinated Hydra Head V1 Specification defines several hypotheses under which the security properties defined hold. Any assessment performed during this audit must be done under these hypotheses.
+
+FIXME: there are currently no hypotheses defined in the specification. See section 6.1.2 of the paper for instance
+> Our protocol gives different security guarantees depending on the level of adversarial corruption. It provides correctness independently of both, the number of corrupted head parties and the network conditions. But the guarantee that the protocol makes progress (i.e., that new transactions get confirmed in the head) is only provided in the case that no head parties are corrupted and that the network conditions are good.
+
+In addition to these hypotheses, the following hypotheses are to be considered about the running environment while assessing the Hydra node implementation::
+* The cardano-node and the hydra-node software both run on the same machine;
+* The cardano-node and the hydra-node software communicate through local unix socket;
+* The cardano-node software is trusted;
+* The hydra-node software is trusted;
+* The hydra-node API is only accessible from the local machine;
+* For any client software connecting to the hydra-node through its API, the hydro-node and the client software both run on the same machine;
+* The local machine running a hydra-node is trusted;
+* The hydra and cardano signing keys storage and operation are trusted.
+* For any legitimate hydra-node, the network is resilient enough that it can synchronize the cardano blockchain at regular interval smaller than any head contestation period they participate in.
```

Comment:
This means: we are connected to the cardano network at least once per contestation period?

Reply:
Yes.
I'd like to stick to network hypotheses as formulated in the original paper or in head V1 but I must admit that, as of now, I can't manage to understand where this information is described in either of these documents :(

I found that in spec V1 but that does not look like what I'm looking for as it does not mention any time contraint:

> The head protocol gets correctly (and with completeness) notified about observed on-chain transactions belonging to the respective head instance.

Reply:
We keep that one even if it's not perfect but better statement welcomed.